fdm-core/src/authentication.ts

```diff
@@ -0,0 +1,119 @@
+import { betterAuth } from "better-auth"
+import { drizzleAdapter } from "better-auth/adapters/drizzle"
+import { grantRole } from "./authorization"
+import * as authNSchema from "./db/schema-authn"
+import { handleError } from "./error"
+import type { FdmType } from "./fdm"
+
+export type BetterAuth = ReturnType<typeof betterAuth>
+
+export function createFdmAuth(fdm: FdmType): BetterAuth {
+    // Validate all required environment variables upfront
+    const googleClientId = getRequiredEnvVar("GOOGLE_CLIENT_ID")
+    const googleClientSecret = getRequiredEnvVar("GOOGLE_CLIENT_SECRET")
+    const msClientId = getRequiredEnvVar("MS_CLIENT_ID")
+    const msClientSecret = getRequiredEnvVar("MS_CLIENT_SECRET")
+
+    const auth: BetterAuth = betterAuth({
+        database: drizzleAdapter(fdm, {
+            provider: "pg",
+            schema: authNSchema,
+        }),
+        user: {
+            additionalFields: {
+                firstname: {
+                    type: "string",
+                    required: false,
+                    defaultValue: null,
+                },
+                surname: {
+                    type: "string",
+                    required: false,
+                    defaultValue: null,
+                },
+                lang: {
+                    type: "string",
+                    required: true,
+                    defaultValue: "nl-NL",
+                },
+                farm_active: {
+                    type: "string",
+                    required: false,
+                    defaultValue: null,
+                },
+            },
+        },
+        session: {
+            expiresIn: 60 * 60 * 24 * 30, // 30 days
+            updateAge: 60 * 60 * 24, // 1 day (every 1 day the session expiration is updated)
+        },
```

Comment:
this duration interval is common best practice or just a first guess?

Reply:
The default for expiresIn is 7 days, but I thought 30 days is more convenient for users. Maybe for security it is better to reduce to 7 days? https://www.better-auth.com/docs/concepts/session-management#session-expiration

Reply:
difficult to judge. since is does not require manual actions of the user, you might switch to 7.